File: src/main/javabak/machine_abstractions/FluidCapableBlockEntity.java
package redcrafter07.processed.block.machine_abstractions;

import net.minecraft.world.level.block.state.BlockState;
import net.neoforged.neoforge.fluids.capability.IFluidHandler;

import javax.annotation.Nullable;

public interface FluidCapableBlockEntity {
    @Nullable
    IFluidHandler fluidCapabilityForSide(@Nullable BlockSide side, BlockState state);
}

File: src/main/kotlin/redcrafter07/processed/block/machine_abstractions/FluidCapableBlockEntity.kt
package redcrafter07.processed.block.machine_abstractions

import net.minecraft.world.level.block.state.BlockState
import net.neoforged.neoforge.fluids.capability.IFluidHandler

interface FluidCapableBlockEntity {
    fun fluidCapabilityForSide(side: BlockSide?, state: BlockState): IFluidHandler?
}